sql/parser/eval_test.go

```diff
@@ -233,21 +233,22 @@ func TestEval(t *testing.T) {
 		// Cast expressions.
 		{`true::boolean`, `true`},
 		{`true::int`, `1`},
-		{`true::float`, `1`},
+		{`true::float`, `1.0`},
 		{`length(true::text)`, `4`},
 		{`false::boolean`, `false`},
 		{`false::int`, `0`},
-		{`false::float`, `0`},
+		{`false::float`, `0.0`},
 		{`length(false::text)`, `5`},
 		{`1::boolean`, `true`},
 		{`0::boolean`, `false`},
 		{`1::int`, `1`},
-		{`1::float`, `1`},
+		{`1::float`, `1.0`},
 		{`length(123::text)`, `3`},
 		{`1.1::boolean`, `true`},
 		{`0.0::boolean`, `false`},
 		{`1.1::int`, `1`},
 		{`1.1::float`, `1.1`},
+		{`1000000::float`, `1000000.0`},
```

Comment:
Ah, yeah, this will have to be in TestEval. TestParse keeps everything represented as `NumVal`, which is just a string.